src/PhpEnvironment/Request.php

```diff
@@ -233,7 +233,7 @@ public function setServer(ParametersInterface $server)
             }
         }
 
-        $this->getHeaders()->addHeaders($headers);
+        $headers = $this->getHeaders()->addHeaders($headers);
```

Comment:
Personally I am not sure if it is good improvement, I mean, for me it is less readable - as definition of the variable and usage of it is several lines below.
Also it takes advantages of fluent interfaces, and as far as I know we tend to drop them.

Reply:
Ok, I reverted the change of re-assignment headers.